packages/odata-common/src/request-builder/batch/batch-request-builder.ts

```diff
@@ -38,15 +38,25 @@ export class BatchRequestBuilder<
       | GetAllRequestBuilderBase<EntityBase, DeSerializersT>
       | GetByKeyRequestBuilderBase<EntityBase, DeSerializersT>
       | Omit<
-          ActionFunctionImportRequestBuilderBase<unknown, ODataRequestConfig>,
+          ActionFunctionImportRequestBuilderBase<
+            DeSerializersT,
+            unknown,
+            ODataRequestConfig
+          >,
           'execute'
         >
     )[]
   ) {
     super(new ODataBatchRequestConfig(defaultBasePath));
-    this.deSerializers = first(
-      Object.values(this.getEntityToApiMap())
-    )?.deSerializers;
+
+    const entityApi = first(Object.values(this.getEntityToApiMap()));
+    const entityDeserializer = entityApi?.deSerializers;
+
+    const operationDeserializer = this.getOperationDeserializer(requests);
+
+    this.deSerializers = entityDeserializer
+      ? entityDeserializer
+      : operationDeserializer;
```

Comment:
[pp] This can be shortened quite a bit and I think would be appropriate here as it makes it slightly more readable in my opinion.

```suggestion
    this.deSerializers = entityApi?.deSerializers || this.getOperationDeserializer(requests);
```